app/logic/spreadsheet.py

```diff
@@ -0,0 +1,515 @@
+#query data and generate spreadsheet
+from os import major
+from app.models.eventParticipant import EventParticipant
+from app.models.user import User
+from app.models.programEvent import ProgramEvent
+from app.models.program import Program
+from app.models.event import Event
+from app.models.term import Term
+from peewee import *
+import openpyxl
+import pandas as pd
+from openpyxl.utils import get_column_letter
+from openpyxl.worksheet.table import Table, TableStyleInfo
+
+
+def volunteerHoursByProgram():
+    query = (ProgramEvent.select(ProgramEvent.program_id, fn.SUM(EventParticipant.hoursEarned).alias('sum'))
+                         .join(EventParticipant, on=(ProgramEvent.event == EventParticipant.event))
+                         .join(Program, on=(Program.id == ProgramEvent.program_id))
+                         .group_by(ProgramEvent.program_id))
+
+    totalHoursByProgram= {pe.program.programName: float(pe.sum) for pe in query}
+    return totalHoursByProgram
+
+def volunteerHoursAllPrograms():
+    totalHoursAllProgram = float(EventParticipant.select(fn.Sum(fn.Coalesce(EventParticipant.hoursEarned, 0))).scalar())
+    return totalHoursAllProgram
+
+
+def volunteersMajors():
+    # Majors represented in volunteering
+    majorQuery = (User.select(User.major, fn.COUNT(fn.DISTINCT(EventParticipant.user_id)).alias('count'))
+                      .join(EventParticipant, on=(User.username == EventParticipant.user_id))
+                      .group_by(User.major))
+    major_query_dict = {row.major: float(row.count) for row in majorQuery}
+
+    return major_query_dict
+
+def classLevelsInVolunteering():
```

Comment:
Combine class level and major queries into one function where you pass through which one you want to get.